src/interface/metadata.hpp

```diff
@@ -325,14 +327,16 @@ class Metadata {
       return Edge;
     } else if (IsSet(Node)) {
       return Node;
+    } else if (IsSet(Particle)) {
+      return Particle;
```

Comment:
OK I see your point about topological location. And since as you point out below we can just use `IsSet(Particle)` in `metadata.cpp` this is unnecessary -- I've removed this